shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/route/engine/validator/dml/impl/ShardingInsertStatementValidator.java

```diff
@@ -97,5 +97,17 @@ public void postValidate(final ShardingRule shardingRule, final SQLStatementCont
         if (needCheckDatabaseInstance) {
             Preconditions.checkState(routeContext.isSingleRouting(), "Sharding value must same with subquery.");
         }
+        if (routeContext.isSingleRouting()) {
```

Comment:
@chengh1 Please add test case for this change.

Reply:
The integration test module already contains my test scenarios, so I don’t need to add a new test case.